### DISCUSSION TITLE
Daily psalms change in Officium Parvum B.M.V.

### DISCUSSION THREAD

**Discussion by Augustinus-Altovadensis** [DivinumOfficium/divinum-officium#2534] [Q&A]
Pax!
I would like to introduce the correct psalms into the Officium Parvum B.M.V. It requires the following:
1. Remove Psalm 66 from the Lauds. The rest of the festive psalms remains.
2. Introduce the psalm changing schema. It works basically so, that you wouldn't duplicate the daily psalms in the Little Office. 
3. Keep the special incipit (half of _Ave Maria_, followed by _Deus, in adjutórium_.) and conclusion (after the collect just the _Dulce nomen_.)

**Ad 1.** I have tried to copy the approach used in other horas and made the [Special Lauds] section in the appropriate _C12.txt_ files. Unfortunately this doesn't work and the [Special Lauds] section doesn't work at all. Which is a shame, unlike the day horas, the Lauds are at least same every day...
**Ad 2.** Adding a new section in the _Psalmi minor.txt_ file seems to be a suitable approach. Unfortunately, when using the [Special Prima], the _psalmi_minor()_ routine from _specials.pl_ doesn't seem to run with the "Specials" keyword used.
**Ad 3.** This works fine with the [Special $hora] section, but this creates the already mentioned problem that the psalms are not imported...

Does anyone have any hints? Thanks a lot!

**Comment by mbab** [comment]
> **Ad 1.** I have tried to copy the approach used in other horas and made the [Special Lauds] section in the appropriate _C12.txt_ files. Unfortunately this doesn't work and the [Special Lauds] section doesn't work at all. Which is a shame, unlike the day horas, the Lauds are at least same every day...

In `specials.pl`
```
my $i = ($hora =~ /laudes/i) ? ' 2' : ($hora =~ /vespera/i) ? " $vespera" : '';
if (exists($w{"Special $hora$i"})) { return loadspecial($w{"Special $hora$i"}); }
```

This look like section should be called `[Special Laudes 2]`.

> 2. Introduce the psalm changing schema. It works basically so, that you wouldn't duplicate the daily psalms in the Little Office.

What is that schema?

**Reply by Augustinus-Altovadensis** [reply]
Thanks for the reply!
Yes, that makes sense when looking at the code... But I wouldn't have guessed it.
I still needed to account for the schema for the minor hours, so I made new `*Parvum.txt` files in the `Ordinarium` folder, which along with a new entry in the `Psalmi minor.txt` file and slight changes in the code did the trick. I'm not that happy about changing the code, but I don't think I did have a choice.

The schema is following: 
**Sunday:** minor hours with psalms from rest of the week (with small exception in Prime)
**Monday, Wednesday, Friday:** minor hours with psalms from Sunday
**Tuesday, Thursday, Saturday:** minor hours with psalms from Monday
More can be found [here](http://www.gregorianbooks.com/gregorian/www/www.kellerbook.com/PARVUM~2.HTM) under `Cistercian 1930`, but I was following the actual Cistercian liturgical books.

Anyway, this topic has been solved and it works fine.
